src/wp-includes/script-loader.php

```diff
@@ -265,15 +268,23 @@ function wp_register_development_scripts( $scripts ) {
  * @param WP_Scripts $scripts WP_Scripts object.
  */
 function wp_default_packages_scripts( $scripts ) {
-	$suffix = defined( 'WP_RUN_CORE_TESTS' ) ? '.min' : wp_scripts_get_suffix();
+	$suffix      = wp_scripts_get_suffix();
+	$assets_path = ABSPATH . WPINC . "/assets/script-loader-packages{$suffix}.php";
+	// In the case when the development files are missing (e.g. when running unit tests),
+	// we fallback to the production version of assets that are always present.
+	if ( ! $suffix && ! is_readable( $assets_path ) ) {
+		$suffix      = '.min';
+		$assets_path = ABSPATH . WPINC . '/assets/script-loader-packages.min.php';
```

Comment:
Looks like this hard-codes a presumption about the suffix. While the presumption is correct, might it be better for the application stability to only have that hard-coded value in one place ?

While `wp_scripts_get_suffix()` already allows for a `$type` parameter, it doesn't currently allow for that `$type` to be set to `'prod'`/`'nodev'`, so there may be room for improvement there ?
(The function also doesn't document the valid value(s) which can be passed, so the docs for the function definitely need to be improved. Additionally, the "There are two suffix types, the normal one and the dev suffix." line in the docs also seem misleading with respect to what the function does.)

Alternatively, it could be considered to set a `WP_SCRIPT_SUFFIX_MIN` constant to `'.min'` and use that in all the applicable places.
Something along those lines may even already exist... (I haven't checked).

Reply:
> Alternatively, it could be considered to set a `WP_SCRIPT_SUFFIX_MIN` constant to `'.min'` and use that in all the applicable places. Something along those lines may even already exist... (I haven't checked).

I didn't find anything like that. I named the constant `WP_ASSET_SUFFIX_MIN` in case we want to use it with CSS, too. See [bd7297d](https://github.com/WordPress/wordpress-develop/pull/3340/commits/bd7297dae6ba9ec0942c9a326c540198b76c5eba). We probably could refactor more code, unrelated to the changes in this PR.